apps/judicial-system/backend/src/app/modules/file/file.service.ts

```diff
@@ -71,11 +92,74 @@ export class FileService {
     })
   }
 
-  private async throttleUploadStream(
+  private getFileProperties(file: CaseFile, theCase: Case) {
+    let courtDocumentFolder: CourtDocumentFolder
+    let subject: string
+
+    switch (file.category) {
+      case CaseFileCategory.COVER_LETTER:
+        courtDocumentFolder = CourtDocumentFolder.INDICTMENT_DOCUMENTS
+        subject = this.formatMessage(courtUpload.coverLetter, {
+          courtCaseNumber: theCase.courtCaseNumber,
+        })
+        break
+      case CaseFileCategory.INDICTMENT:
+        courtDocumentFolder = CourtDocumentFolder.INDICTMENT_DOCUMENTS
+        subject = this.formatMessage(courtUpload.indictment, {
+          courtCaseNumber: theCase.courtCaseNumber,
+        })
+        break
+      case CaseFileCategory.CRIMINAL_RECORD:
+        courtDocumentFolder = CourtDocumentFolder.INDICTMENT_DOCUMENTS
+        subject = this.formatMessage(courtUpload.criminalRecord, {
+          courtCaseNumber: theCase.courtCaseNumber,
+        })
+        break
+      case CaseFileCategory.COST_BREAKDOWN:
+        courtDocumentFolder = CourtDocumentFolder.INDICTMENT_DOCUMENTS
+        subject = this.formatMessage(courtUpload.coustBreakdown, {
+          courtCaseNumber: theCase.courtCaseNumber,
+        })
+        break
+      case CaseFileCategory.COURT_RECORD:
+        courtDocumentFolder = CourtDocumentFolder.COURT_DOCUMENTS
+        subject = this.formatMessage(courtUpload.courtRecord, {
+          courtCaseNumber: theCase.courtCaseNumber,
+        })
+        break
+      case CaseFileCategory.RULING:
+        courtDocumentFolder = CourtDocumentFolder.COURT_DOCUMENTS
+        subject = this.formatMessage(courtUpload.verdict, {
+          courtCaseNumber: theCase.courtCaseNumber,
+        })
+        break
+      case CaseFileCategory.CASE_FILE_CONTENTS:
+        courtDocumentFolder = CourtDocumentFolder.CASE_DOCUMENTS
+        subject = this.formatMessage(courtUpload.caseFileContents, {
+          courtCaseNumber: theCase.courtCaseNumber,
+        })
+        break
+      case CaseFileCategory.CASE_FILE:
+        courtDocumentFolder = CourtDocumentFolder.CASE_DOCUMENTS
+        subject = this.formatMessage(courtUpload.caseFile, {
+          courtCaseNumber: theCase.courtCaseNumber,
+        })
+        break
+      default:
+        courtDocumentFolder = CourtDocumentFolder.CASE_DOCUMENTS
+        subject = file.name
+    }
+
+    const fileName = file.category
+      ? file.name.replace(/^.+\./, `${subject}.`)
```

Comment:
Consider moving the regex into a variable with a descriptive name for better code readability

Reply:
Makes sense. Will do.